### PR TITLE
feat: implementing the chat ops UI and subscription with apollo client

### PR DESCRIPTION
**Pull request by duvanmonsa**
- Adding apollo client and sse link client for subscriptions 
- Required libraries to parse MD on the chat response
- Implement the new Chat UI components 
- Implement new subscription and query required to consume platform.

### CODE REVIEW

File: plugins/stateful/src/apollo.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        { ...operation, query: print(operation.query) },
+        {
+          next: sink.next.bind(sink),
+          complete: sink.complete.bind(sink),
+          error: sink.error.bind(sink),
+        },
+      );
+    });
+  }
+}
```
Comment (peraltafederico) [inline plugins/stateful/src/apollo.ts]: @duvanmonsa I thought SSE should work by default with @apollo/client, similar as how it does with RedwoodJS. I guess `graphql-sse` is needed because it doesn't work, but why?

File: plugins/stateful/src/apollo.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      };
+    },
+  })
+
+  const authLink = setContext(async () => {
+    return {
+      headers: {
+        authorization: `Bearer ${token}`,
+        'Auth-Provider': 'auth0',
+      },
```
Comment (peraltafederico) [inline plugins/stateful/src/apollo.ts]: @duvanmonsa Move this to a common object

File: plugins/stateful/src/apollo.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    };
+  })
+
+  const splitLink = split(
+    ({ query }) => {
+      const definition = getMainDefinition(query);
+      return (
+        definition.kind === 'OperationDefinition' &&
+        definition.operation === 'subscription'
+      );
```
Comment (peraltafederico) [inline plugins/stateful/src/apollo.ts:69]: @duvanmonsa Why is this needed? Leave a comment explaining this code.

File: plugins/stateful/src/App.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  defaultOptions: {
+    queries: {
+      retry: false,
+    },
+  },
+});
+
+export const App = () => {
+  return (
+    <QueryClientProvider client={queryClient}>
```
Comment (peraltafederico) [inline plugins/stateful/src/App.tsx:17]: @duvanmonsa Move both providers `react-query` and `apollo-client` here

**Review by peraltafederico — COMMENTED** [review]
@duvanmonsa Make sure `prettier:check` passes. I saw some files that need it!

File: plugins/stateful/src/apollo.ts (hunk 1 above)
Comment (duvanmonsa) [reply plugins/stateful/src/apollo.ts]: SSE is required because apollo does not know anything about SSE.

File: plugins/stateful/src/apollo.ts (hunk 3 above)
Comment (duvanmonsa) [reply plugins/stateful/src/apollo.ts:69]: That's documented in apollo docs is a normal pattern.
https://www.apollographql.com/docs/react/data/subscriptions

File: plugins/stateful/src/App.tsx (hunk 4 above)
Comment (duvanmonsa) [reply plugins/stateful/src/App.tsx:17]: Apollo-client requires react-query to be ready that's why they can't be together at least for now.

**Review by peraltafederico — APPROVED** [review]
